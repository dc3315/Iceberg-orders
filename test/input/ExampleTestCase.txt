B,100323,99,50000
B,100324,98,25500
S,100325,100,500
S,100326,100,100000,10000
S,100327,103,100
S,100328,105,20000
B,100329,100,16000
