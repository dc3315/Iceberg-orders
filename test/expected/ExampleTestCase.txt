+-----------------------------------------------------------------+
| BUY                            | SELL                           |
| Id       | Volume      | Price | Price | Volume      | Id       |
+----------+-------------+-------+-------+-------------+----------+
|    100323|       50,000|     99|       |             |          |
+----------+-------------+-------+-------+-------------+----------+

+-----------------------------------------------------------------+
| BUY                            | SELL                           |
| Id       | Volume      | Price | Price | Volume      | Id       |
+----------+-------------+-------+-------+-------------+----------+
|    100323|       50,000|     99|       |             |          |
|    100324|       25,500|     98|       |             |          |
+----------+-------------+-------+-------+-------------+----------+

+-----------------------------------------------------------------+
| BUY                            | SELL                           |
| Id       | Volume      | Price | Price | Volume      | Id       |
+----------+-------------+-------+-------+-------------+----------+
|    100323|       50,000|     99|    100|          500|    100325|
|    100324|       25,500|     98|       |             |          |
+----------+-------------+-------+-------+-------------+----------+

+-----------------------------------------------------------------+
| BUY                            | SELL                           |
| Id       | Volume      | Price | Price | Volume      | Id       |
+----------+-------------+-------+-------+-------------+----------+
|    100323|       50,000|     99|    100|          500|    100325|
|    100324|       25,500|     98|    100|       10,000|    100326|
+----------+-------------+-------+-------+-------------+----------+

+-----------------------------------------------------------------+
| BUY                            | SELL                           |
| Id       | Volume      | Price | Price | Volume      | Id       |
+----------+-------------+-------+-------+-------------+----------+
|    100323|       50,000|     99|    100|          500|    100325|
|    100324|       25,500|     98|    100|       10,000|    100326|
|          |             |       |    103|          100|    100327|
+----------+-------------+-------+-------+-------------+----------+

+-----------------------------------------------------------------+
| BUY                            | SELL                           |
| Id       | Volume      | Price | Price | Volume      | Id       |
+----------+-------------+-------+-------+-------------+----------+
|    100323|       50,000|     99|    100|          500|    100325|
|    100324|       25,500|     98|    100|       10,000|    100326|
|          |             |       |    103|          100|    100327|
|          |             |       |    105|       20,000|    100328|
+----------+-------------+-------+-------+-------------+----------+

100329,100325,100,500
100329,100326,100,15500
+-----------------------------------------------------------------+
| BUY                            | SELL                           |
| Id       | Volume      | Price | Price | Volume      | Id       |
+----------+-------------+-------+-------+-------------+----------+
|    100323|       50,000|     99|    100|        4,500|    100326|
|    100324|       25,500|     98|    105|       20,000|    100328|
|          |             |       |    103|          100|    100327|
+----------+-------------+-------+-------+-------------+----------+

